core/trino-parser/src/main/java/io/trino/sql/tree/JsonPathParameter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class JsonPathParameter
+        extends Node
+{
+    private final Identifier name;
+    private final Expression parameter;
+    private final Optional<JsonFormat> format;
+
+    public JsonPathParameter(Identifier name, Expression parameter, Optional<JsonFormat> format)
```

Comment:
Same as the earlier question. Where is the variant without location used?

Reply:
Same answer as https://github.com/trinodb/trino/pull/9831#discussion_r827857020